lvsfunc/comparison.py

```diff
@@ -27,111 +27,64 @@ class Direction(IntEnum):
+        self.num_clips = len(clips)
 
-            for key, (name, clip) in enumerate(self.clips.items()):
-                top_crop = crop_height * key
-                bottom_crop = crop_height * (self.num_clips - 1 - key)
+        widths: Set[int] = {clip.width for clip in self.clips}
+        self.width: Optional[int] = w if (w := widths.pop()) != 0 and len(widths) == 0 else None
 
-                if key != (self.num_clips - 1):
-                    bottom_crop += overflow
+        heights: Set[int] = {clip.height for clip in self.clips}
+        self.height: Optional[int] = h if (h := heights.pop()) != 0 and len(heights) == 0 else None
 
-                self.clips[name] = clip.std.Crop(top=top_crop, bottom=bottom_crop)
+        formats: Set[Optional[str]] = {getattr(clip.format, 'name', None) for clip in self.clips}
+        self.format: Optional[str] = formats.pop() if len(formats) == 0 else None
 
-    def _markclips(self) -> List[vs.VideoNode]:
+    def _markclips(self) -> None:
         """
-        This allows for internal functions to specify a space-only string,
-        such as '   ' that avoid placing a label on the clip.
-
-        By default, `vapoursynth.core.text.Text(text='')` will not place a label,
-        but specifying multiple clips in the **clips Dict with the same '' key will not work.
+        If a `name` is only space characters, `'   '`, for example, the name will not be overlayed on the clip.
         """
-        clips: List[vs.VideoNode] = []
-        for name, clip in self.clips.items():
-            if name.strip():
-                clips.append(clip.text.Text(text=name, alignment=self.label_alignment))
-            else:
-                clips.append(clip)
-
-        return clips
+        if self.names:
+            self.clips = [clip.text.Text(text=name, alignment=self.label_alignment) if name.strip() else clip for clip, name in zip(self.clips, self.names)]
 
     @abstractmethod
     def _compare(self) -> vs.VideoNode:
         raise NotImplementedError
 
+    @property
     def clip(self) -> vs.VideoNode:
         """
         Returns the comparison as a single VideoNode for further manipulation or attribute inspection.
 
-        `comp_clip = Comparer(**kwargs, name1=clip1, name2=clip2, ...).clip()` is the intended use in encoding scripts.
+        `comp_clip = Comparer(...).clip` is the intended use in encoding scripts.
         """
         return self._compare()
 
-    def compare(self) -> None:
+    def set_output(self) -> None:
         """
         Directly calls ``vapoursynth.VideoNode.set_output()`` on the :py:class:`lvsfunc.comparison.Comparer` generated comparison clip.
 
-        `Comparer(**kwargs, name1=clip1, name2=clip2, ...).compare()` is the intended use in encoding scripts linked with a previewer.
+        `Comparer(...).set_output()` is the intended use in encoding scripts linked with a previewer.
         """
         return self._compare().set_output()
```

Comment:
This function seems pretty redundant to me when it's equivalent to `.clip.set_output()`.

Reply:
Script writers are going to be used to doing `compare(clipa, clipb).set_output()` (probably) and this seemed like a nice convenient way of still being able to do that without using the clip return in-between.

Reply:
Seems a bit dishonest to kind of pretend you have a clip but only implement `set_output()`. If the usage of `compare` is important, then that should just return the Comparer's `clip` immediately.